src/USER-ESFRICTION/fix_esfriction.cpp

```diff
@@ -0,0 +1,203 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+       Contributing Author: Harsh Hemani and Manoj Warrier
+       (Bhabha Atomc Research Center, Computational Analysis Division,
+        Visakhapatnam, India)
+       harshscience777@gmail.com, Manoj.Warrier@gmail.com
+---------------------------------------------------------------------- */
+
+#include <mpi.h>
+#include <cmath>
+#include <cstring>
+#include "fix_esfriction.h"
+#include "atom.h"
+#include "domain.h"
+#include "update.h"
+#include "error.h"
+#include "force.h"
+#include "respa.h"
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+FixESFriction::FixESFriction(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg)
+{
+  if (narg < 4) error->all(FLERR,"Illegal fix es-friction command");
+  // args -- 0: fix, 1: fix-id, 2: group-id
+  if (strcmp(update->unit_style,"metal") != 0) {
+    error->all(FLERR, "fix es-friction supports only metal units");
+  }
+  Z1 = utils::numeric(FLERR, arg[3], false, lmp); // atomic number of PKA
+  Z2 = utils::numeric(FLERR, arg[4], false, lmp); // atomic number of target
+  A1 = utils::numeric(FLERR, arg[5], false, lmp); // mass number of PKA
+  A2 = utils::numeric(FLERR, arg[6], false, lmp); // mass number of target
+  Ne_val = utils::numeric(FLERR, arg[7], false, lmp);  // no. of valance shell electrons
+  E_pka = utils::numeric(FLERR, arg[8], false, lmp);   // PKA energy
+  /* ES_cutoff is the energy cutoff for applying ES.
+     That is, atoms with energy below this will not experience ES */
+  ES_cutoff = utils::numeric(FLERR, arg[9], false, lmp);
+  // Compute total energy lost due to ES as per NRT model
+  double E_hat = get_ehat();
+  ES_loss_NRT = E_pka - E_hat;
+  // following computes coefficient of friction (beta) using LS model
+  double number_density = atom->natoms / 
+    (domain->xprd * domain->yprd * domain->zprd) * 1e30; // number-per-m3
+  double E_0 = h_cross*h_cross / (2*m_e) * 
+    pow( 3*pi*pi*Ne_val*(number_density), 2.0/3.0) * joules_to_ev; // eV
```

Comment:
Please use the `MY_PI` constant from the `MathConst` namespace to have a consistent use and precision of the PI constant and not your own definition.